jaxley/morphology/morph_utils.py

```diff
@@ -0,0 +1,105 @@
+# This file is part of Jaxley, a differentiable neuroscience simulator. Jaxley is
+# licensed under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+import networkx as nx
+import numpy as np
+
+from jaxley.io.graph import connect_graphs, from_graph, to_graph
+
+
+def morph_delete(module_view) -> "Cell":
+    """Deletes part of a morphology.
+
+    Args:
+        module_view: View of a `jx.Module`. Defines the compartments to be deleted.
+
+    Returns:
+        A cell in which specified compartments are deleted.
+
+    Example:
+    --------
+
+    ::
+
+        cell = jx.read_swc("path_to_swc_file.swc", ncomp=1)
+        cell = morph_delete(cell.axon)
+    """
+    # If the user did not run `compute_xyz` or `compute_compartment_centers`, we run
+    # it automatically.
+    if np.isnan(module_view.base.xyzr[0][0, 0]):
+        module_view.base.compute_xyz()
+    if "x" not in module_view.base.nodes.columns:
+        module_view.base.compute_compartment_centers()
+
+    comps_to_delete = module_view.nodes.index
+    comp_graph = to_graph(module_view.base)
+
+    nodes_to_keep = []
+    for node in comp_graph.nodes:
+        # We keep all `branchpoints`. The ones that have degree <2 will automatically
+        # be trimmed away `_remove_branch_points_at_tips`, which is run during
+        # `from_graph()`.
+        if (
+            comp_graph.nodes[node]["type"] == "branchpoint"
+            or comp_graph.nodes[node]["comp_index"] not in comps_to_delete
+        ):
+            nodes_to_keep.append(node)
+
+    comp_graph = nx.subgraph(comp_graph, nodes_to_keep)
+    cell = from_graph(comp_graph)
```

Comment:
variable name `cell` could be a bit confusing since it should work on any view no?

Reply:
if it only works for cells, we should assert that there is only a single cell index in nodes or sth.